### PR TITLE
kubevirt, e2e: Replace kv client with cr client

### PR DESCRIPTION
**Pull request by qinqon**
**- What this PR does and why is it needed**
The KubeVirt client is introducing k8s 1.26 as dependency and that makes bumping to k8s 1.28 impossible. This change replace the kv client with controller runtime client and the run commands is using virtctl bash command.

closes https://issues.redhat.com/browse/CNV-34370

**- Description for the changelog**
Remove kubevirt client from test/e2e golang depedencies.

### CODE REVIEW

File: contrib/kind.sh [hunk 1]
```diff
@@ -1309,6 +1309,7 @@ function is_nested_virt_enabled() {
 }
 
 function install_kubevirt() {
+        local kubevirt_version="v1.0.0"
```
Comment (maiqueb) [inline contrib/kind.sh:1312]: while I agree this makes sense (IMHO tracking nightly builds was too aggressive) can you elaborate why you're doing this *now* ? 

And if you don't mind, move that into its own commit.

**Review by maiqueb — CHANGES_REQUESTED** [review]
Thank you for taking this up.

I just have a request to split the PR in two commits, since IMO it looks two things are being done here:
- install KubeVirt v1.0.0 instead of tracking the kubevirt main branch
- replace KubeVirt's client-go w/ controller-runtime + virtctl

I assume we would prefer to use KubeVirt's client-go; thus, once KubeVirt project bumps to 1.28, we could (should ?) revert the latter commit. And, I honestly think we should pin a tagged version instead of tracking the latest main branch. 

Thoughts ?

File: contrib/kind.sh [hunk 2]
```diff
@@ -1336,6 +1336,29 @@ function install_kubevirt() {
...
+    pushd ./bin
+       if [ ! -f ./virtctl ]; then
+           cli_name="virtctl-${kubevirt_version}-${OS_TYPE}-${ARCH}"
+           curl -LO "${kubevirt_release_url}/${cli_name}"
+           mv ${cli_name} virtctl
+           if_error_exit "Failed to download virtctl failed!"
+       fi
+    popd
+
+    chmod +x ./bin/virtctl
```
Comment (maiqueb) [inline contrib/kind.sh:1361]: IMO this is part of the other commit - where you stop using kubevirt's client-go.

**Review by maiqueb — CHANGES_REQUESTED** [review]
Thanks for splitting the commits. I think they're wrong though.

**Review by maiqueb — APPROVED** [review]
Thank you !!!

**Comment by tssurya** [comment]
EFW flake: https://github.com/ovn-org/ovn-kubernetes/issues/3970

File: contrib/kind.sh [hunk 3]
```diff
@@ -1336,6 +1336,29 @@ function install_kubevirt() {
...
         done
     fi
+    
+    if [ ! -d "./bin" ]
+    then
+        mkdir -p ./bin
+        if_error_exit "Failed to create bin dir!"
+    fi
+
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
```
Comment (tssurya) [inline contrib/kind.sh:1346]: I guess these two are the only types we support right? and so its not really a new dependency and I think we call `install_kubevirt` only if the kubevirt flag is set so we are good on the default cases...

File: contrib/kind.sh (hunk 2 above)
Comment (tssurya) [inline contrib/kind.sh:1361]: so if user is running the kind script locally on laptop, then when we delete the cluster using the destroy do we need to clean up this somehow? or its ok?

**Review by tssurya — APPROVED** [review]
overall lgtm, this definitely unblocks me on https://github.com/ovn-org/ovn-kubernetes/pull/3978
left a few questions..

File: contrib/kind.sh (hunk 3 above)
Comment (qinqon) [reply contrib/kind.sh:1346]: The code is an "adaptation" of https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.sh#L29, at the end is the same thing different binary.

File: contrib/kind.sh (hunk 2 above)
Comment (qinqon) [reply contrib/kind.sh:1361]: This is the same as "./bin/kubectl" and it's local to the repo I think we are fine.

File: contrib/kind.sh [hunk 4]
```diff
@@ -1336,6 +1336,29 @@ function install_kubevirt() {
...
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        OS_TYPE="darwin"
+    fi
+
+    pushd ./bin
+       if [ ! -f ./virtctl ]; then
+           cli_name="virtctl-${kubevirt_version}-${OS_TYPE}-${ARCH}"
+           curl -LO "${kubevirt_release_url}/${cli_name}"
+           mv ${cli_name} virtctl
+           if_error_exit "Failed to download virtctl failed!"
```
Comment (jcaamano) [inline contrib/kind.sh:1357]: nit: `Failed to download virtctl`

**Comment by jcaamano** [comment]
Now the kv webhook failed even after 10 retries
https://github.com/ovn-org/ovn-kubernetes/actions/runs/6729252031/job/18290427615?pr=3981
